src/poetry/core/version/markers.py

```diff
@@ -630,28 +623,16 @@ def of(cls, *markers: BaseMarker) -> BaseMarker:
                     for i, mark in enumerate(new_markers):
                         if isinstance(mark, SingleMarker) and (
                             mark.name == marker.name
-                        ):
-                            # Markers with the same name have the same constraint type,
-                            # but mypy can't see that.
-                            constraint_union = mark.constraint.union(
-                                marker.constraint  # type: ignore[arg-type]
+                            or (
+                                mark.name in PYTHON_VERSION_MARKERS
+                                and marker.name in PYTHON_VERSION_MARKERS
                             )
-                            if constraint_union == mark.constraint:
-                                included = True
-                                break
-                            elif constraint_union == marker.constraint:
-                                new_markers[i] = marker
-                                included = True
-                                break
-                            elif constraint_union.is_any():
-                                return AnyMarker()
-                            elif (
-                                isinstance(constraint_union, VersionConstraint)
-                                and constraint_union.is_simple()
-                            ):
-                                new_markers[i] = SingleMarker(
-                                    mark.name, constraint_union
-                                )
+                        ):
+                            new_marker = _merge_single_markers(mark, marker, cls)
+                            if new_marker is not None:
+                                if new_marker.is_any():
```

Comment:
It can save some loop cycles and function calls, but in the big picture, I think marker simplifications don't add up to a significant part of the overall time. So let's remove it for now until someone does some real world measurements.